pymongo/mongo_client.py

```diff
@@ -836,6 +837,8 @@ def __init__(
             # This will be used later if we fork.
             MongoClient._clients[self._topology._topology_id] = self
 
+        self._indefinite_error: Optional[WriteConcernError] = None
```

Comment:
We can't use an instance variable to track this because it makes MongoClient not thread safe.